framework/src/bcs/IntegratedBC.C

```diff
@@ -104,9 +104,10 @@ IntegratedBC::IntegratedBC(const InputParameters & parameters)
     MooseVariable * var = &_subproblem.getVariable(_tid, _diag_save_in_strings[i]);
 
     if (var->feType() != _var.feType())
-      mooseError("Error in " + name() + ". When saving diagonal Jacobian values in an Auxiliary "
-                                        "variable the AuxVariable must be the same type as the "
-                                        "nonlinear variable the object is acting on.");
+      paramError(
+          "diag_save_in",
+          "saved-in auxiliary variable is incompatible with the object's nonlinear variable: ",
+          moose::internal::incompatVarMsg(*var, _var));
```

Comment:
Does this method really belong in the moose::internal namespace? It looks like it's something that should be exposed to end users.

Reply:
BTW - I'm supportive of migrating to better name spacing in the long run. We'll have to come up with a way to support the transition. Perhaps we can do this by creating the new namespace with the desired declarations and calling methods in the existing namespace?

Reply:
I don't anticipate users often writing code that needs to check for moosevariable type compatibility.  And even if they do, I think the chances that they know about and use this method (or that we notice and inform them of it) are pretty low.  If we see specific cases where those assumptions begin to be violated, it is easy to pull it out of "internal", but not as much the other way around.